config/main.py

```diff
@@ -2994,7 +2994,14 @@ def _qos_update_ports(ctx, ports, dry_run, json_data):
         for table_name in tables_multi_index:
             entries = config_db.get_keys(table_name)
             for key in entries:
-                port, _ = key
+                # Add support for chassis/multi-dut:
+                # on a single-dut, key = ('Ethernet136', '6')
+                # while on a chassis, key = ('str2-chassis-lcx-1', 'Asic0', 'Ethernet84', '5')
+                for element in key:
+                    if element.startswith('Eth'):
+                        port = element
+                        break
+                assert port is not None, "Port is not found in config DB"
```

Comment:
please remove assert, instead display a CLI message

Reply:
revisited in https://github.com/sonic-net/sonic-utilities/pull/3234